File: app/src/main/java/com/commit451/viewholderarrayadapter/sample/CheeseSpinnerAdapter.java
package com.commit451.viewholderarrayadapter.sample;

import android.content.Context;
import android.support.annotation.NonNull;
import android.view.ViewGroup;

import com.commit451.viewholderarrayadapter.ViewHolderArrayAdapter;

import java.util.List;

/**
 * Example usage of {@link ViewHolderArrayAdapter}
 */
public class CheeseSpinnerAdapter extends ViewHolderArrayAdapter<Cheese, CheeseViewHolder, CheeseDropDownViewHolder> {

    public CheeseSpinnerAdapter(Context context, List<Cheese> objects) {
        super(context, objects);
    }

    @NonNull
    @Override
    protected CheeseViewHolder onCreateViewHolder(@NonNull ViewGroup parent) {
        return CheeseViewHolder.inflate(parent);
    }

    @NonNull
    @Override
    protected CheeseDropDownViewHolder onCreateDropDownViewHolder(@NonNull ViewGroup parent) {
        return CheeseDropDownViewHolder.inflate(parent);
    }

    @Override
    protected void onBindViewHolder(@NonNull CheeseViewHolder holder, Cheese item, int position) {
        holder.bind(item);
    }

    @Override
    protected void onBindDropDownViewHolder(@NonNull CheeseDropDownViewHolder holder, Cheese item, int position) {
        holder.bind(item);
    }
}

File: app/src/main/java/com/commit451/viewholderarrayadapter/sample/CheeseSpinnerAdapter.kt
package com.commit451.viewholderarrayadapter.sample

import android.content.Context
import com.commit451.viewholderarrayadapter.ViewHolderArrayAdapter

/**
 * Example usage of [ViewHolderArrayAdapter]
 */
class CheeseSpinnerAdapter(
    context: Context,
    objects: List<Cheese>
) : ViewHolderArrayAdapter<Cheese, CheeseViewHolder, CheeseDropDownViewHolder>(
    context = context, objects = objects,
    onCreateViewHolder = { parent ->
        CheeseViewHolder.inflate(parent)
    },
    onCreateDropDownViewHolder = { parent ->
        CheeseDropDownViewHolder.inflate(parent)
    },
    onBindViewHolder = { viewHolder, _, item ->
        viewHolder.bind(item)
    },
    onBindDropDownViewHolder = { viewHolder, _, item ->
        viewHolder.bind(item)
    }
)
